site/src/components/Button/Button.tsx

```diff
@@ -51,9 +51,12 @@ export const Button: FC<ButtonProps> = forwardRef<
 	const Comp = asChild ? Slot : "button";
 	return (
 		<Comp
-			className={cn(buttonVariants({ variant, size, className }))}
 			ref={ref}
 			{...props}
+			className={cn(buttonVariants({ variant, size, className }))}
+			type={
+				props.type === undefined && Comp === "button" ? "button" : props.type
+			}
```

Comment:
Small change to make sure that you have to opt into submit behavior for true buttons

Reply:
I’m not sure I fully understand the value or use case for this 🤔. Could you share an example to help clarify? At first glance, it seems like the type prop should be the consumer’s responsibility rather than something built into the component.

Reply:
So, I feel that +90% of the time, when you have a `<button>`, you want it to be `type="button"` instead of the default `type="submit"`, and even experienced frontend engineers can forget to do that. That sometimes leads to forms accidentally getting submitted when you didn't mean for them to. I think it's a more reasonable default value that can't ever be made official because of HTML backwards-compatibility

And since the Radix button component is polymorphic, we don't have any linting rules to catch that you didn't set the type, but also because it's polymorphic, the final output might not be an actual button. If we use `asChild` to style a link like a button, we can't always assume that we can attach a `type` attribute. `<a>` doesn't support the prop, so while React would let us do it no problem, the HTML would be invalid

So basically, the check makes it so that buttons will default to `type="button"` if the output would be a button, and if the prop wasn't already set. Otherwise, yeah, it does fall to the user to wire things up correctly

Reply:
Do you think it would make sense to summarize this decision and add it as a comment? I’m concerned we might forget this decision in the future and accidentally remove it, as it’s not a very common or usual approach.